packages/core/src/Docs/Domain/DocumentBodyModel.ts

```diff
@@ -319,21 +328,25 @@ export class DocumentBodyModel extends DocumentBodyModelSimple {
         this.deleteTree(nodes, currentIndex, textLength);
     }
 
-    getParagraphByTree(nodes: DataStreamTreeNode[], insertIndex: number): Nullable<DataStreamTreeNode> {
+    private getParagraphByTree(nodes: DataStreamTreeNode[], insertIndex: number): Nullable<DataStreamTreeNode> {
         for (const node of nodes) {
-            const { startIndex, endIndex, children } = node;
+            const { children } = node;
+
             if (node.exclude(insertIndex)) {
                 continue;
             }
+
             if (node.nodeType === DataStreamTreeNodeType.PARAGRAPH) {
                 return node;
             }
+
             return this.getParagraphByTree(children, insertIndex);
         }
+
         return null;
     }
 
-    foreachTop(
+    private foreachTop(
```

Comment:
A good habit to narrow interfaces! 👍

Reply:
The name is better to be `forEachTop` like the native `forEach`.

Reply:
+1